lib/engine/game/g_system18/map_gotland_customization.rb

```diff
@@ -0,0 +1,535 @@
+            GSystem18::Step::BuyTrain,
+          ]
+        end
+
+        def map_gotland_init_round
+          # Start with difficulty selection if not already set
+          return GSystem18::Round::DifficultySelection.new(self, [GSystem18::Step::DifficultySelection]) unless difficulty_level
+
+          @log << "-- #{round_description('Stock', 1)} --"
+          @round_counter = 1
+          stock_round
+        end
+
+        def reduce_float_costs(floated_corporation_index)
+          order = @corporation_float_order
+          if floated_corporation_index&.positive? && floated_corporation_index&.<(order.size)
+            order = order[floated_corporation_index..-1] + order[0...floated_corporation_index]
+          end
+          count = 0
+          order.each do |corp_hash|
+            corp = corporation_by_id(corp_hash[:sym])
+            next if corp.floated?
+
+            ability = corp.all_abilities.find { |a| a.is_a?(Engine::Ability::FloatCost) && a.float_cost&.positive? }
+            new_cost = difficulty_level_value * count
+            count += 1
+            if new_cost.zero?
+              corp.remove_ability(ability) if ability
+            else
+              ability ||= Engine::Ability::FloatCost.new(type: 'float_cost', description: '', desc_detail: '',
+                                                         float_cost: new_cost)
+              corp.add_ability(ability) unless corp.all_abilities.include?(ability)
+              ability.float_cost = new_cost
+              ability.description = "Float cost: #{new_cost}"
+              ability.desc_detail = "This corporation requires #{new_cost} to float"
+            end
+          end
+        end
+
+        def can_par?(corporation, player)
+          player_president_shares = corporations.count { |corp| corp.president?(player) }
+          return false if player_president_shares >= 4
+
+          super
+        end
+
+        def float_corporation(corporation)
+          player = corporation.owner
+
+          ability = corporation.all_abilities.find do |a|
+            a.is_a?(Engine::Ability::FloatCost) && a.float_cost && a.float_cost.positive?
+          end
+          if ability
+            float_cost = ability.float_cost
+            if player.cash < float_cost
+              raise GameError,
+                    "#{player.name} does not have enough cash to float #{corporation.name}. "
+            end
+            player.spend(float_cost, @bank)
+            @log << "#{player.name} pays #{format_currency(float_cost)} to float #{corporation.name}"
+            corporation.remove_ability(ability)
+          end
+          floated_index = @corporation_float_order.index { |c| c[:sym] == corporation.id }
+          reduce_float_costs(floated_index)
+          super
+          # Track newly floated corporations for the current stock round
+          @round.track_newly_floated(corporation) if @round.is_a?(GSystem18::Round::GotlandStock)
+        end
```

Comment:
map_gotland_float_corporation instead. You will have to add the framework to the game.rb file. That way future maps can provide their own float_corporation.